backend/inspirehep/editor/cache.py

```diff
@@ -0,0 +1,80 @@
+from datetime import datetime
+
+import structlog
+from flask import current_app
+from redis import StrictRedis
+
+from inspirehep.config import EDITOR_LOCK_EXPIRATION_TIME
+
+LOGGER = structlog.getLogger()
+
+
+class EditorSoftLock(object):
+    def __init__(self, recid, record_version, user_email=None, task_name=None):
+        if not (task_name or user_email):
+            raise ValueError("User mail or task name must be passed")
+        if user_email:
+            self.is_user_lock = True
+            self.key = user_email
+        else:
+            self.is_user_lock = False
+            self.key = task_name
+        self.recid = recid
+        self.record_version = record_version
+
+    @property
+    def redis(self):
+        redis_url = current_app.config.get("CACHE_REDIS_URL")
+        redis = StrictRedis.from_url(redis_url)
+        return redis
+
+    @property
+    def hash_name(self):
+        if self.is_user_lock:
+            return self._get_hash_name_for_edited_record()["user"]
+        return self._get_hash_name_for_edited_record()["task"]
+
+    @property
+    def message(self):
+        if self.is_user_lock:
+            return f"{self.key} at {datetime.now().isoformat()}"
+        return f"{self.key} at {datetime.now().isoformat()}"
+
+    def _get_hash_name_for_edited_record(self):
+        return {
+            "user": f"editor-lock:{self.recid}@{self.record_version}",
+            "task": f"editor-task-lock:{self.recid}@{self.record_version}",
+        }
+
+    def add_lock(self):
+        self.redis.hset(self.hash_name, key=self.key, value=self.message)
+        self.redis.expire(self.hash_name, time=EDITOR_LOCK_EXPIRATION_TIME)
```

Comment:
FYI, redis is non transactional by default, so a different task/user might come in between these two actions and do some other things. I don't think it's a problem in this specific case, but something to always keep in mind when you're making several logically related changes to redis.